### PR TITLE
add internal/cache/gache/option test

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I added `internal/cache/gache/option` test and updated comment for each function.

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/cache/gache/option_test.go [hunk 1]
```diff
@@ -19,462 +19,372 @@ package gache
...
 			if test.beforeFunc != nil {
 				test.beforeFunc(test.args)
 			}
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+			got := defaultOptions()
```
Comment (hlts2) [inline internal/cache/gache/option_test.go:97]:
```suggestion
			type args struct{}
			type want struct {
				want *cache
			}
			type test struct {
				name       string
				args       args
				want       want
				checkFunc  func(want, *cache) error
				beforeFunc func(args)
				afterFunc  func(args)
			}

			defaultCheckFunc := func(w want, cache) error {
				// please check field with go-cmp
				// e.g.) https://github.com/vdaas/vald/blob/feature/internal/tensorflow_warmup/internal/core/converter/tensorflow/tensorflow_test.go#L51-L70
				return nil
			}
			g := new(cache)
			for _, opt := range defaultOpts() {
				if err := opt(g); err != nil {
					tt.Fatal(err)
				}
			}
			if err := test.checkFunc(test.want, g); err != nil {
				tt.Errorf("error = %v", err)
			}
```

Comment (vankichi) [reply internal/cache/gache/option_test.go:97]: @hlts2 
what is `defaultOpts()` ? 🤔 
I cannot find the definition about it 😓

File: internal/cache/gache/option.go [hunk 2]
```diff
@@ -26,18 +26,21 @@ import (
 
 type Option func(*cache)
```
Comment (kevindiu) [inline internal/cache/gache/option.go:27]: I think we also need comment for this type

**Pull request merged**